pkg/controller/certificates/internal/secretsmanager/secret.go

```diff
@@ -203,8 +206,24 @@ func (s *SecretsManager) setValues(crt *cmapi.Certificate, secret *corev1.Secret
 			delete(secret.Data, jksSecretKey)
 			delete(secret.Data, jksTruststoreKey)
 		}
-	}
 
+		// Add additional key format for the key
+		if utilfeature.DefaultFeatureGate.Enabled(feature.ExperimentalCertificatePrivateKeyAdditionalFormats) {
+			if crt.Spec.PrivateKey.AdditionalKeyFormats != nil {
+				for _, f := range crt.Spec.PrivateKey.AdditionalKeyFormats {
+					switch f.Type {
+					case cmapi.PrivateKeyFormatDER:
+						block, _ := pem.Decode(data.PrivateKey)
+						secret.Data["key.der"] = block.Bytes
+					case cmapi.PrivateKeyFormatPEM:
+						secret.Data["tls.pem"] = []byte(fmt.Sprintf("%s%s", data.PrivateKey, data.Certificate))
```

Comment:
Are we sure we'll handle new-lines here correctly? `strings.Join([]string{data.PrivateKey, data.Certificate}, "\n")` may ensure we handle this more correctly in edge-cases where the private key doesn't end with a newline :)

Reply:
Using String.join instead of fmt.Sprintf with "\n" at the end to be sure.